File: hw/tipantiza/u3/hw26TemplateMethodPattern/TemplateMethod_js/controller/BeverageController.js
const Tea = require('../model/Tea');
const Coffee = require('../model/Coffee');
const Chocolate = require('../model/Chocolate');

class BeverageController {
    async prepareAllBeverages() {
        const tea = new Tea();
        const coffee = new Coffee();
        const chocolate = new Chocolate();

        console.log("\nMaking tea ...");
        await tea.prepareRecipe();

        console.log("\nMaking coffee ...");
        await coffee.prepareRecipe();

        console.log("\nMaking chocolate ...");
        await chocolate.prepareRecipe();
    }
}

module.exports = BeverageController;

File: hw/basurto/u3/hw25TemplateMethod/hw25TemplateMethod.py/controller/beverage_controller.py
from model.tea import Tea
from model.coffee import Coffee

class BeverageController:
    def __init__(self, view):
        self.view = view

    def prepare_tea(self):
        self.view.show_message("\nMaking tea ...")
        tea = Tea()
        # Modifica el comportamiento del hook dinámicamente usando la Vista
        tea.wants_condiments = lambda: self.view.get_user_input("Would you like lemon with your tea (y/n)?").lower().startswith("y")
        tea.prepare_recipe()

    def prepare_coffee(self):
        self.view.show_message("\nMaking coffee ...")
        coffee = Coffee()
        # Modifica el comportamiento del hook dinámicamente usando la Vista
        coffee.wants_condiments = lambda: self.view.get_user_input("Would you like milk and sugar with your coffee (y/n)?").lower().startswith("y")
        coffee.prepare_recipe()